src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/LincheckJavaAgent.kt

```diff
@@ -344,12 +344,17 @@ internal object LincheckClassFileTransformer : ClassFileTransformer {
 
     override fun transform(
         loader: ClassLoader?,
-        internalClassName: String,
+        internalClassName: String?,
         classBeingRedefined: Class<*>?,
         protectionDomain: ProtectionDomain?,
         classBytes: ByteArray
     ): ByteArray? = runInIgnoredSection {
         // If the class should not be transformed, return immediately.
+        if (classBeingRedefined == null) {
```

Comment:
Could you please explain what `if (classBeingRedefined == null)` means?

Reply:
That the class, being instrumented is null.
Here is the source of the failure. It happens because instrumentation does not a class.
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/5fae1c64-90b2-41eb-8444-61a636ce0ee9">
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/6fd18b30-0e1a-466b-bfc0-28f650f94f9e">